examples/testing/README.md

```diff
@@ -0,0 +1,105 @@
+Instructions For Running The Testing Examples
+=============================================
+
+The examples included here are used as automated tests, but they can also be explored manually.
+The fisrt example is very simple and subsequent examples are progressively more complex,
+so you can use these examples as an introduction into configuring StoneWork.
+
+It is recommended to read the [Getting started examlpe][getting-started] before proceeding to these examples.
+
+This document describes common prerequisities and instructions for running the examples.
+Specific information about individual examples can be found in their respective README files.
+
+Prerequisities
+--------------
+
+1. 
+   Make sure that `stonework` production image and testing-only image `tester` are already built.
+   In the top-level directory of the repository, trigger the makefile target `images` with:
+   ```
+   $ make images
+   ```
+2. 
+   **Optional**: StoneWork can be managed through `agentctl`, a CLI provided by the Ligato framework.
+   You can install it with
+   ```
+   $ go get go.ligato.io/vpp-agent/v3/cmd/agentctl
+   ```
+   This will enable you to use `agentctl` to access StoneWork directly from the host machine.
+
+   If you skip this step, you will still be able to use `docker` to access `agentctl`, that comes pre-installed in the `stonework` container.
+
+Running The Examples: The Automated Way
+---------------------------------------
+
+The examples can be easily controlled using their respective Makefile.
+
+Just navigate to the directory of the chosen example and run `make test`
+to run the example, perform automated tests and dump logs in case of failure.
+
+Alternatively, to have more control over the execution of the example, you can use these commands:
+
+- First, run `make start-example` to deploy the containers and apply the startup configuration.
+- When the containers are running, run `make test-stonework` to run automated tests.
+- When the containers are running, run `make dump-logs` to dump StoneWork logs.
+  The logs will be saved into a file `example.log` (if the file exists, it will be overwritten).
+- Finally, shut down and clean up everything (except dumped logs) with `make stop-example`.
+
+Running The Examples: The Manual Way
```

Comment:
Might be a bit cleaner to have section `Running Examples` as header level 2 and all the ways to run them as separate sub-section with header level 3. This will help to generate better more structured TOC (e.g. in case it gets converted to HTML format).